src/main/scala/org/ergoplatform/api/ApiCodecs.scala

```diff
@@ -98,7 +98,7 @@ trait ApiCodecs {
 
   // TODO catena: this should be removed (Value should not be used directly)
   implicit val valueEncoder: Encoder[Value[SType]] = { value =>
-    ErgoTreeSerializer.DefaultSerializer.serializeWithSegregation(value).asJson
+    ValueSerializer.serialize(value).asJson
```

Comment:
@greenhat Isn't it a step back? From explicit ErgoTree serialization back to implicit value serialization.

Reply:
I don't think so. `valueEncoder` used extensively (additional registers encoder, context extension encoder, in tx's `toString`, etc.). I see no way to use `ErgoTreeSerializer` since it requires SigmaProp values. Previously it did cast any value as SigmaProp. I've tried to remove `valueEncoder`completely and got into the "rabbit hole" quickly.

Reply:
1) additional registers contain only constants
2) context extensions also contain only constants
Usage of Value[SType] here basically means "I don't know which values this can be, so let it be any possible value.".
It is better to have more concrete encoders (actually only Constants is needed) than single universal encoder, that just hide all potential problems.

Reply:
Done in https://github.com/ergoplatform/ergo/pull/681/commits/240cd33d25a23771323d59019f41c3cdd38ac4cf